cli/deployment/api_client.go

```diff
@@ -11,73 +11,233 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+type Status int
+
+const (
+	Done     Status = 0
+	Pending  Status = 1
+	Running  Status = 2
+	NotFound Status = 3
+	Error    Status = -1
+)
+
+const (
+	ActionApply  string = "apply"
+	ActionDelete string = "delete"
+	ActionCreate string = "create"
+	ActionUpdate string = "update"
+)
+
+var actions = []string{ActionApply, ActionDelete, ActionCreate, ActionUpdate}
+
+// Function runGCloud exposed to variable in order to mock it inside api_client_test.go
 // The runGCloud function runs the gcloud tool with the specified arguments. It is implemented
 // as a variable so that it can be mocked in tests of its exported consumers.
 var runGCloud = func(args ...string) (result string, err error) {
-	args = append(args, "--format", "yaml")
 	log.Println("gcloud", strings.Join(args, " "))
 	cmd := exec.Command("gcloud", args...)
-	outBuff := &bytes.Buffer{}
-	errBuff := &bytes.Buffer{}
-	cmd.Stdout = outBuff
-	cmd.Stderr = errBuff
 	// pass user's PATH env variable, expected gcloud executable can be found in PATH
 	cmd.Env = append(cmd.Env, fmt.Sprintf("PATH=%s", os.Getenv("PATH")))
 
-	if err := cmd.Start(); err != nil {
-		log.Printf("Failed to start cmd: %v, \n Output:\n %v", err, errBuff.String())
-		return "", err
-	}
+	output, err := cmd.CombinedOutput()
 
-	if err := cmd.Wait(); err != nil {
-		log.Printf("Cmd returned error: %v, \n Output:\n %v", err, errBuff.String())
-		return errBuff.String(), err
+	if err != nil {
+		log.Printf("failed to start cmd: %v, output: %v", err, string(output))
+		return string(output), err
 	}
 
-	return outBuff.String(), err
+	return string(output), err
 }
 
 // GetOutputs retrive existing Deployment outputs using gcloud and store result in map[string]string
 // where "resourceName.propertyName" is key, and value is string representation of the output value.
```

Comment:
The type returned by this function has changed, so please update the documentation comment for this function.